internal/wasm/jit/asm/amd64/impl.go

```diff
@@ -0,0 +1,382 @@
+package amd64
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/tetratelabs/wazero/internal/wasm/jit/asm"
+)
+
+// nodeImpl implements asm.Node for amd64.
+type nodeImpl struct {
+	instruction asm.Instruction
+
+	// offsetInBinary represents the offset of this node in the final binary.
+	offsetInBinary int64
+	// jumpTarget holds the target node in the linked for the jump-kind instruction.
+	jumpTarget *nodeImpl
+	// prev holds the previous node to this node in the assembled linked list.
+	prev *nodeImpl
+	// next holds the next node from this node in the assembled linked list.
+	next *nodeImpl
+
+	types                    operandTypes
+	srcReg, dstReg           asm.Register
+	srcConst, dstConst       int64
+	srcMemIndex, dstMemIndex asm.Register
+	srcMemScale, dstMemScale byte
+
+	mode byte
+}
+
+// AssignJumpTarget implements asm.Node.AssignJumpTarget.
+func (n *nodeImpl) AssignJumpTarget(target asm.Node) {
+	n.jumpTarget = target.(*nodeImpl)
+}
+
+// AssignSourceConstant implements asm.Node.AssignSourceConstant.
+func (n *nodeImpl) AssignDestinationConstant(value int64) {
+	n.dstConst = value
+}
+
+// AssignSourceConstant implements asm.Node.AssignSourceConstant.
+func (n *nodeImpl) AssignSourceConstant(value int64) {
+	n.srcConst = value
+}
+
+// OffsetInBinary implements asm.Node.OffsetInBinary.
+func (n *nodeImpl) OffsetInBinary() int64 {
+	return n.offsetInBinary
+}
+
+// String implements fmt.Stringer.
```

Comment:
add a comment below this to say what format we are expected, maybe a simple example and a link